src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,205 @@
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.ModuleReflectionUtil;
+
+public class XdocsUrlTest {
+
+    private static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    private static final String TREE_WORKER = "TreeWalker";
+
+    private static final String SUFFIX_CHECK = "Check";
+
+    private static final String CHECKS = "checks";
+
+    private static final String MISC = "misc";
+
+    private static final String ANNOTATION = "annotation";
+
+    private static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    private static final String INDENTATION = "Indentation";
+
+    private static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    private static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, ArrayList<String>> getXdocsMap() throws IOException {
+        final Map<String, ArrayList<String>> checksNamesMap = new HashMap<>();
+        final Set<Class<?>> checkSet = CheckUtil.getCheckstyleModules();
+        final Set<Class<?>> treeWalkerCheckSet = checkSet.stream()
+                .filter(ModuleReflectionUtil::isCheckstyleTreeWalkerCheck)
+                .collect(Collectors.toSet());
+        final Set<Class<?>> fileSetCheckSet = checkSet.stream()
+                .filter(ModuleReflectionUtil::isFileSetModule)
```

Comment:
Can't we `or` the filter stream to combine the 2 checks, isCheckstyleTreeWalkerCheck and isFileSetModule? This way we only need to parse it once and only have 1 set.

Reply:
done